src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy

```diff
@@ -328,4 +331,27 @@ class UnityPlugin implements Plugin<Project> {
             task.group = GROUP
         }
     }
+
+    private static void addAddUPMPackagesTask(Project project, final UnityPluginExtension extension) {
+        def addUPMPackagesTask = project.tasks.register(Tasks.addUPMPackages.toString(), AddUPMPackages) {task ->
+            task.group = GROUP
+        }
+        def manifestFile = extension.projectDirectory.map({ it.file("Packages/manifest.json") })
+
+        addUPMPackagesTask.configure( { t->
+            t.manifestPath.set(manifestFile)
+            t.upmPackages.putAll(extension.getUPMPackages())
+
+            // should be made from the Test task setup?
```

Comment:
No, here is way better. However, we still need to add a dependency to this task on the test one.